contrib/platform/src/com/sun/jna/platform/win32/Guid.java

```diff
@@ -466,16 +468,14 @@ public CLSID(GUID guid) {
 
     /**
      * The Class REFIID.
-     *
-     * @author Tobias Wolf, wolf.tobias@gmx.net
      */
-    public class REFIID extends IID {
+    public class REFIID extends ByReference {
```

Comment:
My reasoning for the com.sun.jna.ptr.ByReference approach was that it is closest to the underlying definition. It is clear by that no magic handling is expected from JNA, just passing the pointer.

If this is strictly out of line - I only see the Option to make the members of the GUID structure final so that when passed in they are only read and in general GUID must be interpreted as a value type.